desk

```diff
@@ -114,56 +114,95 @@ cmd_run() {
 }
 
 
+# Usage:        desk list [options]
+# Description:  List all desks along with a description
+# --only-names: List only the names of the desks
+# --no-format:  Use ' - ' to separate names from descriptions
 cmd_list() {
     if [ ! -d "${DESKS}/" ]; then
-        echo "No desk dir! Run 'desk init'." 
+        echo "No desk dir! Run 'desk init'."
         exit 1
     fi
+    local SHOW_NAMES DESKEXT AUTO_ALIGN name desc len longest out
+
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+            --only-names) test "$SHOW_NAMES" != false && SHOW_DESCRIPTIONS=false && AUTO_ALIGN=false ;;
+            --no-format) AUTO_ALIGN=false ;;
+        esac
+        shift
+    done
 
-    local DESKEXT=$(get_deskfile_extension)
-                       
-    find "${DESKS}/" -name "*${DESKEXT}" -print0 | while read -d '' -r f; do
-        local name=$(basename "${f/${DESKEXT}//}") 
-        local desc=$(echo_description "$f")
+    DESKEXT=$(get_deskfile_extension)
+    out=""
+    longest=0
 
-        if [ -z "$desc" ]; then
-            echo "$name"
+    while read -d '' -r f; do
+        name=$(basename "${f/${DESKEXT}//}")
+        if [[ "$SHOW_DESCRIPTIONS" = false ]]; then
+            out+="$name"$'\n'
         else
-            echo "$name" "-" "$desc"
+            desc=$(echo_description "$f")
+            out+="$name - $desc"$'\n'
+            len=${#name}
+            (( len > longest )) && longest=$len
         fi
-    done
+    done < <(find "${DESKS}/" -name "*${DESKEXT}" -print0)
+    if [[ "$AUTO_ALIGN" != false ]]; then
+        print_aligned "$out" "$longest"
+    else
+        printf "%s" "$out"
```

Comment:
Despite the occasional verboseness, I still prefer to use `printf` instead of `echo` virtually everywhere because the output is often much more clearly described by the command itself (i.e. I find it easier to reason about). Here, using `echo` would add an extra newline since `"$out"` already contains one. 

If you'd rather, however, I will make this change.